### ISSUE TITLE
Exception (oxtorrent): A task was canceled.: A task was canceled. 

### ISSUE THREAD

**Issue by neogamaes** [Jackett/Jackett#7573]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue.
Check closed issues as well, because your issue may have already been fixed.

Please read our [Contributing Guidelines](https://github.com/Jackett/Jackett/blob/master/CONTRIBUTING.md) before submitting your issue to ensure a prompt response to your bug.

### Environment

**OS**: NAS DS218j DSM 6.2.2

**.Net Runtime**: [.Net-Core/.Net-Framework/Mono] Mono 

**.Net Version**: Mono 5.18

**Jackett Version**: 0.13.529

**Last Working Jackett Version**:

**Are you using a proxy or VPN?** [yes] a VPN on the NAS

### Description
Hello, 
I am completely new using jackett/sonarr but I had a successful config 2 days ago with about 10 indexer working fine. 
Since the last night only 2 trackers works nyaa and YTS) .. All the others (kickass, horriblesub, 1337x, oxtorrent...) have the message: An error occured while testing this indexer
Exception (oxtorrent): A task was canceled.: A task was canceled.

One of the URL is the following: https://www.oxtorrent.com/

I don't understand why it worked and not anymore. 

I tried reboot few times the NAS (218j) but it changed nothing. 

please help :)

thank you 
### Logged Error Messages
```
Jackett.Common.IndexerException: Exception (oxtorrent): A task was canceled. ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
  at System.Net.Http.HttpClientHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x004f5] in <d79bb9faee264b669b8825270136bfaa>:0 
  at CloudflareSolverRe.ClearanceHandler.SendRequestAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x00090] in <44fc4e0cb3fd4a268a136236a74d835e>:0 
  at CloudflareSolverRe.ClearanceHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x000a4] in <44fc4e0cb3fd4a268a136236a74d835e>:0 
  at System.Net.Http.HttpClient.SendAsyncWorker (System.Net.Http.HttpRequestMessage request, System.Net.Http.HttpCompletionOption completionOption, System.Threading.CancellationToken cancellationToken) [0x000ca] in <d79bb9faee264b669b8825270136bfaa>:0 
  at Jackett.Common.Utils.Clients.HttpWebClient.Run (Jackett.Common.Utils.Clients.WebRequest webRequest) [0x00497] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Common.Utils.Clients.WebClient.GetString (Jackett.Common.Utils.Clients.WebRequest request) [0x0010b] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Common.Indexers.BaseWebIndexer.RequestStringWithCookies (System.String url, System.String cookieOverride, System.String referer, System.Collections.Generic.Dictionary`2[TKey,TValue] headers) [0x000cb] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Common.Indexers.CardigannIndexer.PerformQuery (Jackett.Common.Models.TorznabQuery query) [0x0086a] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery (Jackett.Common.Models.TorznabQuery query) [0x00091] in <81018049d9c64cebaec829a0a1ab9877>:0 
   --- End of inner exception stack trace ---
  at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery (Jackett.Common.Models.TorznabQuery query) [0x000ee] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery (Jackett.Common.Models.TorznabQuery query) [0x0006b] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Common.Services.IndexerManagerService.TestIndexer (System.String name) [0x000a3] in <81018049d9c64cebaec829a0a1ab9877>:0 
  at Jackett.Server.Controllers.IndexerApiController.Test () [0x000e1] in <eae291d0d6ac414ab45f212c1483e5b4>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor+TaskOfIActionResultExecutor.Execute (Microsoft.AspNetCore.Mvc.Infrastructure.IActionResultTypeMapper mapper, Microsoft.Extensions.Internal.ObjectMethodExecutor executor, System.Object controller, System.Object[] arguments) [0x00071] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync () [0x00131] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync () [0x0009e] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow (Microsoft.AspNetCore.Mvc.Filters.ActionExecutedContext context) [0x0001b] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next (Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+State& next, Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker+Scope& scope, System.Object& state, System.Boolean& isCompleted) [0x00382] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync () [0x0002f] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter () [0x0009f] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow (Microsoft.AspNetCore.Mvc.Filters.ResourceExecutedContext context) [0x0001b] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next (Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+State& next, Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker+Scope& scope, System.Object& state, System.Boolean& isCompleted) [0x00840] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync () [0x0002f] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync () [0x0012e] in <b4ef600f4a594fe2865a8f97f915fb9d>:0 
  at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke (Microsoft.AspNetCore.Http.HttpContext httpContext) [0x001cb] in <6092a16d93814eba828b517a2b132f80>:0 
  at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke (Microsoft.AspNetCore.Http.HttpContext context) [0x00384] in <427697fe42b7459ba5302fb76d339d3b>:0 
  at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke (Microsoft.AspNetCore.Http.HttpContext context) [0x0043e] in <f352e566abf6421e87eafbcf57a0b237>:0 
  at Jackett.Server.Middleware.CustomExceptionHandler.Invoke (Microsoft.AspNetCore.Http.HttpContext httpContext) [0x0008a] in <eae291d0d6ac414ab45f212c1483e5b4>:0 
```

**Comment by garfield69** [comment]
cancelled tasks errors happen when Jackett waits for a response from a http GET and gives waiting when there is no reply within the default time limit.
Unfortunately, there is little in the regular stack dump you provided that is useful for diagnosis.
So the first thing to try is to enable the **enhanced** logging, try one of the suspect indexers, and see if the logs provide more useful information.

Please provide the **full enhanced log** file. 
1. scroll down to the bottom of the Jackett Dashboard and *tick* the *enhanced Logging* checkbox
2. scroll up a bit and *click* on the *apply server settings* button
3. repeat the test
4. find the *log.txt* file (linux `~/.config/Jackett/`, windows `%ProgramData%\Jackett`)
5. edit it with a plain text editor, redacting any personal details, usernames, passwords, passkeys, hashes etc.
6. save, then drag-drop it here for us to take a look at.

(Note: When you restart Jackett the enhanced-log switch will be turned off. This prevents performance hits during normal operations).

Thanks.

**Comment by neogamaes** [comment]
Hi,

Thank you for your answer. Please find hereafter the full enhanced log (at least, I think it is :) ) 

[jackett.log](https://github.com/Jackett/Jackett/files/4305468/jackett.log)

**Comment by garfield69** [comment]
no clues l in the log you provided :-(
i don't use proxy and vpn at all so I am not going to much use going forward.
but given that 2 indexers are working and the others are not, I'm fairly confident we can rule out Jackett, other than the fact that you are using mono, which has always been problematic.

Are you able to switch to the .net core version of Jackett, perhaps the armv7 by following the installation notes at https://github.com/Jackett/Jackett#install-on-linux-armv7-or-above
use the LinuxARM32 package for best compatibility

if you get it running and you still have the same problem, then my best guess is that this is an issue with your vpn.
so the next step would be to see if your vpn can be configured to use a different ip block than the one you are currently allocated, to see if that makes a difference.
@cadatoiva  @ngosang  could use your help on this ticket.

**Comment by cadatoiva** [comment]
I'll look at it in the morning with the other one.

**Comment by neogamaes** [comment]
Thank you very much for your support.

What I can try quite fast is to remove VPN, reboot the NAS and see if it changes anything. 

For the stand alone installation of Jackett I will give it a try asap to check if it helps.

**Comment by neogamaes** [comment]
Well.. Good news. Removing the VPN and rebooting the NAS seems to solve the problem. 

Not clear why it was working with the VPN previously.. 

Any idea how I can make it work with VPN ?  Do you think some VPN location can work while some other don't ?

Anyway, on the good way to solve the issue. Thx again.

**Comment by garfield69** [comment]
TLDR; yes definitely.
Details:
most good vpn providers have multiple servers scattered around the world to provide access via  different ISP providers.
The primary purpose is usually for performance, by picking a server close to your locale.
But the Secondary benefit is that different ISPs have different site blocking mechanisms, so choosing another region can help avoid site blocking.
And the third benefit is with the site themselves, which if they are using Cloudflare's anti-DDoS prevention, can blacklist a block of IP addresses as having generated a DDoS attack, and if the VPN server you are using belongs in that IP range you too get blocked. 
Being able to choose an alternate region with you VPN can be helpful.

**Comment by neogamaes** [comment]
Alright, that's clear.
So my understanding is that while I was connected to a given location, my vpn provider change the server IP address (there are indeed several IP for the same location) which was probably blocked by several sites...
I will try several vpn location and find the compromise performance/site blocking.

Thank you. I think we can close the issue... the problem is not linked to Jackett !

Good continuation.

**Comment by cadatoiva** [comment]
The Jackett stand alone has much better performance and fewer bugs, it is worth switching if you are able. We only maintain a Mono version for compatibility with those OS's that cannot use the stand alone version.

And with the  proxy/VPN is there a particular reason why you are trying to use one with Jackett? If you're just trying to avoid DCMA requests from your ISP, then Jackett isn't going to trigger those. You only need to use a proxy if a tracker is geo-blocking (some Italian trackers only accept Italian IP's as example) or if your ISP/country is blocking access to some of the trackers. If you can access all the trackers you use without a proxy, you should consider going without.

**Comment by neogamaes** [comment]
I will try to switch to the stand alone version but if it is working so far I am not sure it's worth it... Unless someone can confirm me it works on the same DSM version I use.

Regarding the VPN, I don't use it especially for Jackett but it is simply activated on my NAS to download peacfuly so I guess it's applied to Jackett as well.. 

If you know how I can have the VPN activated only for Download Station and not for Jackett, then I buy your solution.

**Comment by cadatoiva** [comment]
I don't know much about that platform, so I can't tell you how to be selective with the VPN. I know on my FreeBSD server I can use `setfib` to create virtual networks allowing just my Torrent client to be on the VPN, and everything else in the clear.

Keep the Mono version while it works for you, but know that there are some known bugs on that particular version you might run into.

**Comment by neogamaes** [comment]
Alright, thank you for your help.
